django_cloud_deploy/cli/io.py

```diff
@@ -14,10 +14,75 @@
 """Class for managing console I/O."""
 
 import abc
+import contextlib
 import getpass
 import os
 import re
 import sys
+import threading
+import time
+
+from progress import bar
+
+
+class _ProgressBar(object):
+    """A progress bar showing status of a task.
+
+    Output of the progress bar will be like the following:
+        Processing █████████████████∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙ 55%
+    """
+
+    def __init__(self, expect_time: int, message: str):
+        """Constructor of the class.
+
+        It will take "expect_time" seconds for the progress bar to go from 0%
+        to 100%. If the task takes shorter than "expect time", the progress bar
+        will directly go to 100%.
+
+        Args:
+            expect_time: How long the progress bar is going to run.
+                (In seconds).
+            message: A prefix of the progress bar showing what it is about.
+        """
+        self._expected_time = expect_time
+
+        # The bar will tick every 0.5 seconds.
+        self._bar = bar.ChargingBar(message, max=expect_time * 2)
+        self._thread = threading.Thread(target=self._run)
+        self._bar_lock = threading.Lock()
+
+    def start(self):
+        self._thread.start()
+
+    def finish(self):
+        self._finish()
+
+    def _run(self):
+        """The function to update progress bar."""
+        for _ in range(self._expected_time * 2):
+            self._bar_lock.acquire()
```

Comment:
I think that you can use the lock with a `with` statement

Reply:
Done.